mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -199,11 +200,10 @@ public ConfigBuilder usePigTuples() {
     /**
      * Compute platforms pass down filters to data sources. If the data source cannot apply some filters, or only
      * partially applies the filter, it will return the residual filter back. If the platform can correctly apply
-     * the residual filters, then it should call this api. Otherwise the current api will throw an exception if the
-     * passed in filter is not completely satisfied.
+     * the residual filters, then it should call this api to enable the residual filtering.
      */
-    public ConfigBuilder skipResidualFiltering() {
-      conf.setBoolean(SKIP_RESIDUAL_FILTERING, true);
+    public ConfigBuilder enableResidualFiltering() {
+      conf.setBoolean(SKIP_RESIDUAL_FILTERING, false);
```

Comment:
-1 until this is reverted.

The InputFormat should default to applying residuals because that is what users expect.

Reply:
Reverted.